src/rpc/claimtrie.cpp

```diff
@@ -103,112 +129,27 @@ static bool getValueForOutPoint(const CCoinsViewCache& coinsCache, const COutPoi
-            "    \"claims\": [          (array of object) the claims for this name\n"
-            "      {\n"
-            "        \"claimId\"        (string) the claimId of the claim\n"
-            "        \"txid\"           (string) the txid of the claim\n"
-            "        \"n\"              (numeric) the vout value of the claim\n"
-            "        \"amount\"         (numeric) txout amount\n"
-            "        \"height\"         (numeric) the height of the block in which this transaction is located\n"
-            "        \"value\"          (string) the value of this claim\n"
-            "        \"name\"           (string) the original name of this claim (before normalization)\n"
-            "      }\n"
-            "    ]\n"
-            "  }\n"
-            "]\n");
-
-    LOCK(cs_main);
-
-    CCoinsViewCache coinsCache(pcoinsTip.get());
-    CClaimTrieCache trieCache(pclaimTrie);
-
-    if (!request.params.empty()) {
-        CBlockIndex* blockIndex = BlockHashIndex(ParseHashV(request.params[0], "blockhash (optional parameter 1)"));
-        RollBackTo(blockIndex, coinsCache, trieCache);
-    }
-
-    UniValue ret(UniValue::VARR);
-
-    for (auto it = trieCache.begin(); it != trieCache.end(); ++it) {
-        if (ShutdownRequested())
-            throw JSONRPCError(RPC_INTERNAL_ERROR, "Shutdown requested");
-
-        boost::this_thread::interruption_point();
-
-        if (it->empty())
-            continue;
-
-        UniValue claims(UniValue::VARR);
-        for (const auto& itClaim: it->claims) {
-            UniValue claim(UniValue::VOBJ);
-            claim.pushKV("claimId", itClaim.claimId.GetHex());
-            claim.pushKV("txid", itClaim.outPoint.hash.GetHex());
-            claim.pushKV("n", (int)itClaim.outPoint.n);
-            claim.pushKV("amount", ValueFromAmount(itClaim.nAmount));
-            claim.pushKV("height", itClaim.nHeight);
-            std::string value;
-            if (getValueForOutPoint(coinsCache, itClaim.outPoint, value))
-                ret.pushKV("value", value);
-            else
-                claim.pushKV("error", "TXOUT is spent");
-
-            std::string targetName;
-            CClaimValue targetClaim;
-            if (trieCache.getClaimById(itClaim.claimId, targetName, targetClaim))
-                claim.pushKV("name", targetName);
-
-            claims.push_back(claim);
-        }
-
-        UniValue node(UniValue::VOBJ);
-        node.pushKV("normalized_name", it.key());
-        node.pushKV("claims", claims);
-        ret.push_back(node);
-    }
-    return ret;
+    auto count = params.size();
+    return count == required || count == required + optional;
 }
 
-static UniValue getclaimtrie(const JSONRPCRequest& request)
+static UniValue getclaimsintrie(const JSONRPCRequest& request)
 {
-    if (request.fHelp || request.params.size() > 1)
+    if (request.fHelp || !validParams(request.params, 0, 1))
```

Comment:
not that I am well informed on whats going on here, but @BrannonKing wouldn't it be better to handle this sort of thing in the processing of the request rather than modifying so many lines of code? Seems like a DRY violation.